api/test/unit/utils/schedulerUtils.test.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+var request = require('supertest');
+var expect = require('chai').expect;
+var fs = require('fs');
+var policy = require('../../../lib/models')().policy_json;
+var logger = require('../../../lib/log/logger');
+var nock = require('nock');
+var schedulerUtils = require('../../../lib/utils/schedulerUtils');
+var schedulerURI = process.env.SCHEDULER_URI;
+
+describe('Scheduler Utility functions', function() {
+  var fakePolicy;
+
+  before(function() {
+    fakePolicy = JSON.parse(fs.readFileSync(__dirname+'/../fakePolicy.json', 'utf8'));
+  })
+
+  beforeEach(function() {
+    return policy.truncate();
+  })
+
+  it('should create a new schedule for app id 12345',function(done){
+    nock(schedulerURI)
+    .put('/v2/schedules/12345')
+    .reply(204);
+    var mockRequest = {
+        body : fakePolicy,
+        params : { 'app_id' : '12345' }
+    };
+    schedulerUtils.createOrUpdateSchedule(mockRequest,function(error,result){
+      expect(result.statusCode).to.equal(200);
+      expect(error).to.be.null;
+      done();
+    });
+  });
+
+  it('should fail to create schedule due to error for app id 12346',function(done){
+    var mockError = { 'message':'Failed to create schedules due to an internal' + 
+        ' error in scheduler','details':'fake body' };
+    nock(schedulerURI)
+    .put('/v2/schedules/12346')
+    .replyWithError(mockError);
+    var mockRequest = {
+        body : fakePolicy,
+        params : { 'app_id' : '12346' }
+    };
+    schedulerUtils.createOrUpdateSchedule(mockRequest,function(error,result){
+      expect(error).to.not.be.null;
+      expect(result.statusCode).to.equal(500);
+      expect(error).to.deep.equal(mockError);
+      done();
+    });
+  });
+
+  it('should fail to create a schedule due to internal validation error in scheduler module for app id 12347',function(done){
```

Comment:
Where is the test when an error is returned but no response? Like a dial/connect failure?